### PR TITLE
Bump to v5.5.0

### PR DESCRIPTION
**Pull request by ppeble**
Can merge now!

### CODE REVIEW

**Comment by ppeble** [comment]
@jonathanpike I have this suspicion that we've already discussed this but my brain is mush from sleep deprivation. 😬 

Sorry in advance if we already 'solved' this and I just don't remember.

**Comment by jonathanpike** [comment]
@ppeble sorry it took so long for me to get to this!  I also have a new baby at home (our second), so carving out time for extra projects is sometimes difficult! 

I took a look at the issue and it was... a typo! 👅 I'll have to update the definitions repo, but after that, everything should be good.  I'll do that right now!

**Comment by ppeble** [comment]
@jonathanpike Ha! Not a problem, I'm actually relieved that it is such an easy thing after all of our discussions around this. I was convinced that this was something we already discussed and I just...forgot. 😄 

I'll merge in the def change soon and try to get this out to the door.

Finally: congrats on the new addition to your household! Thanks!

**Comment by ppeble** [comment]
Whoops! Did NOT mean to close this. 😬

**Comment by ppeble** [comment]
😨 

I actually found a real, actual bug in our searching logic that is causing the 'Liberation Day' test to fail. But this time it's actually....me! It's because I reused some variables in a loop and in this specific case it blows stuff up. I'll push a commit.

**Comment by ppeble** [comment]
Look at all that red. 😍

File: test/holidays/finder/context/test_search.rb [hunk 1]
```diff
@@ -194,4 +194,39 @@ def test_returns_unobserved_date_if_observed_method_not_set_but_flag_is_present
       @subject.call(@dates_driver, @regions, @options)
     )
   end
+
+  # This is a specific scenario but it COULD happen in our current flow. The goal: any date
```
Comment (ttwo32) [inline test/holidays/finder/context/test_search.rb]: It's good comment.:smile:

File: lib/generated_definitions/kr.rb [hunk 2]
```diff
@@ -12,18 +12,20 @@ def self.defined_regions
...
-      4 => [{:mday => 8, :function => "lunar_to_solar(year, month, day, region)", :function_arguments => [:year, :month, :day, :region], :name => "Buddah's Birthday", :regions => [:kr]}],
-      5 => [{:mday => 5, :name => "Children's Day", :regions => [:kr]}],
-      6 => [{:mday => 6, :name => "Memorial Day", :regions => [:kr]}],
-      7 => [{:mday => 17, :type => :informal, :name => "Constitution Day", :regions => [:kr]}],
-      8 => [{:mday => 15, :function => "lunar_to_solar(year, month, day, region)", :function_arguments => [:year, :month, :day, :region], :name => "Korean Thanksgiving", :regions => [:kr]},
-            {:mday => 15, :name => "Liberation Day", :regions => [:kr]}],
-      10 => [{:mday => 3, :name => "National Foundation Day", :regions => [:kr]},
-            {:mday => 9, :name => "Hangul Day", :regions => [:kr]}],
-      12 => [{:mday => 25, :name => "Christmas Day", :regions => [:kr]}]
+              1 => [{:mday => 1, :function => "lunar_to_solar(year, month, day, region)", :function_arguments => [:year, :month, :day, :region], :name => "설날", :regions => [:kr]},
```
Comment (ttwo32) [inline lib/generated_definitions/kr.rb]: @ppeble Thank you for merging another pull request of korean definition.:+1:

File: lib/holidays/definition/context/function_processor.rb [hunk 3]
```diff
@@ -9,62 +9,67 @@ def initialize(custom_methods_repo, proc_result_cache_repo)
           @proc_result_cache_repo = proc_result_cache_repo
         end
 
-        def call(year, month, day, func_id, desired_func_args, func_modifier = nil)
-          validate!(year, month, day, func_id, desired_func_args)
+        def call(input, func_id, desired_func_args, func_modifier = nil)
```
Comment (ttwo32) [inline lib/holidays/definition/context/function_processor.rb]: I like this chance. Sometime, many input valiables confused me . 😄

**Comment by ppeble** [comment]
Thanks @ttwo32!

**Pull request merged**